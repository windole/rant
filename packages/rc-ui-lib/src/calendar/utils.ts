/* eslint-disable no-nested-ternary */
export function compareMonth(date1: Date, date2: Date): number {
  const year1 = date1.getFullYear();
  const year2 = date2.getFullYear();

  if (year1 === year2) {
    const month1 = date1.getMonth();
    const month2 = date2.getMonth();
    return month1 === month2 ? 0 : month1 > month2 ? 1 : -1;
  }

  return year1 > year2 ? 1 : -1;
}

export function compareDay(day1: Date, day2: Date): number {
  const compareMonthResult = compareMonth(day1, day2);

  if (compareMonthResult === 0) {
    const date1 = day1.getDate();
    const date2 = day2.getDate();
    return date1 === date2 ? 0 : date1 > date2 ? 1 : -1;
  }

  return compareMonthResult;
}

export const cloneDate = (date: Date): Date => new Date(date);

export const cloneDates = (dates: Date | Date[]): Date | Date[] =>
  Array.isArray(dates) ? dates.map(cloneDate) : cloneDate(dates);

export function getDayByOffset(date: Date, offset: number): Date {
  const cloned = cloneDate(date);
  cloned.setDate(cloned.getDate() + offset);
  return cloned;
}

export const getPrevDay = (date: Date): Date => getDayByOffset(date, -1);
export const getNextDay = (date: Date): Date => getDayByOffset(date, 1);
export const getToday = (): Date => {
  const today = new Date();
  today.setHours(0, 0, 0, 0);
  return today;
};

export function calcDateNum(date: [Date, Date]): number {
  const day1 = date[0].getTime();
  const day2 = date[1].getTime();
  return (day2 - day1) / (1000 * 60 * 60 * 24) + 1;
}

export const getMonthEndDay = (year: number, month: number): number =>
  32 - new Date(year, month - 1, 32).getDate();

const calendarLang = {
  end: '结束',
  start: '开始',
  title: '日期选择',
  confirm: '确定',
  startEnd: '开始/结束',
  weekdays: ['日', '一', '二', '三', '四', '五', '六'],
  monthTitle: (year: number, month: number) => `${year}年${month}月`,
  rangePrompt: (maxRange: number) => `最多选择 ${maxRange} 天`,
};

export const t = (key: string, ...args): string => {
  if (args.length) {
    return calendarLang[key](...args);
  }
  return calendarLang[key];
};

export const formatMonthTitle = (date: Date): string =>
  t('monthTitle', date.getFullYear(), date.getMonth() + 1);
