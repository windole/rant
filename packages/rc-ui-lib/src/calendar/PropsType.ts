import React, { MouseEvent } from 'react';
import { PopupPosition } from '../popup';
import { BaseTypeProps, TeleportType } from '../utils';

export type CalendarType = 'single' | 'range' | 'multiple';

export type CalendarDayType =
  | ''
  | 'start'
  | 'start-end'
  | 'middle'
  | 'end'
  | 'selected'
  | 'multiple-middle'
  | 'multiple-selected'
  | 'disabled'
  | 'placeholder';

export type CalendarDayItem = {
  date?: Date;
  text?: string | number;
  type?: CalendarDayType;
  topInfo?: string;
  className?: string;
  bottomInfo?: string;
};

export type CalendarDayProps = {
  item: CalendarDayItem;
  color: string;
  index: number;
  offset?: number;
  rowHeight: string;
  topInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  bottomInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  onClick?: (item: CalendarDayItem) => void;
};

export type CalendarInstance = {
  reset: (date?: Date | Date[]) => void;
  scrollToDate: (targetDate: Date) => void;
};

export interface CalendarProps extends BaseTypeProps {
  position?: PopupPosition;
  round?: boolean;
  closeOnPopstate?: boolean;
  closeOnClickOverlay?: boolean;
  safeAreaInsetBottom?: boolean;
  teleport?: TeleportType;
  show?: boolean;
  type?: CalendarType;
  title?: React.ReactNode | string;
  subtitle?: React.ReactNode | string;
  color?: string;
  readonly?: boolean;
  poppable?: boolean;
  maxRange?: string | number;
  showMark?: boolean;
  showTitle?: boolean;
  formatter?: (item: CalendarDayItem) => CalendarDayItem;
  rowHeight?: string | number;
  confirmText?: string;
  rangePrompt?: string;
  lazyRender?: boolean;
  showConfirm?: boolean;
  defaultDate?: Date | Date[] | null;
  allowSameDay?: boolean;
  showSubtitle?: boolean;
  showRangePrompt?: boolean;
  confirmDisabledText?: string;
  minDate?: Date;
  maxDate?: Date;
  firstDayOfWeek?: number;
  footer?: React.ReactNode | string;
  topInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  bottomInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  onClickSubtitle?: (ev: MouseEvent) => void;
  /** 点击确认后的回调 */
  onConfirm?: (v: Date | Date[]) => void;
  onOverRange?: () => void;
  onSelect?: (v: Date | Date[]) => void;
  onUnselect?: (v: Date | Date[]) => void;
  onMonthShow?: (obj: { date: Date; title: string }) => void;
  onClose?: () => void;
}

export type CalendarMonthProps = {
  date?: Date;
  type?: CalendarType;
  color?: string;
  minDate?: Date;
  maxDate?: Date;
  showMark?: boolean;
  rowHeight?: number | string;
  formatter?: (item: CalendarDayItem) => CalendarDayItem;
  formatMonthTitle?: (date: Date) => React.ReactNode;
  lazyRender?: boolean;
  currentDate?: Date | Date[];
  allowSameDay?: boolean;
  showSubtitle?: boolean;
  showMonthTitle?: boolean;
  firstDayOfWeek?: number;
  topInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  bottomInfoRender?: (day: CalendarDayItem) => React.ReactNode;
  onClick?: (item: CalendarDayItem) => void;
};

export type CalendarMonthInstance = {
  showed?: boolean;
  getTitle: () => string | React.ReactNode;
  getHeight: () => number;
  setVisible: (value?: boolean | undefined) => void;
  scrollToDate: (body: Element, targetDate: Date) => void;
};
