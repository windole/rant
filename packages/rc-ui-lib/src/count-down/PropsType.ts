import React from 'react';
import { CurrentTime } from '../hooks/use-count-down';
import { BaseTypeProps } from '../utils';

export interface CountDownProps extends BaseTypeProps {
  /** 	倒计时时长，单位毫秒 */
  time?: number | string;
  /** 时间格式	 */
  format?: string;
  /**  是否自动开始倒计时 */
  autoStart?: boolean;
  /**  是否开启毫秒级渲染 */
  millisecond?: boolean;
  /**  倒计时结束时触发 */
  onFinish?: () => void;
  /**  是否开启毫秒级渲染 */
  onChange?: (value: CurrentTime) => void;
  renderChildren?: (value: CurrentTime) => React.ReactNode;
}

export type CountDownCurrentTime = CurrentTime;

export type CountDownInstance = {
  start: () => void;
  pause: () => void;
  reset: () => void;
};
