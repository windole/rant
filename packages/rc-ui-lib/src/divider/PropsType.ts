import React from 'react';
import { BaseTypeProps } from '../utils';

export type DividerContentPosition = 'left' | 'center' | 'right';
export interface DividerProps extends BaseTypeProps, React.HTMLAttributes<HTMLDivElement> {
  /** 是否使用虚线	 */
  dashed?: boolean;
  /**
   * 是否使用 0.5px 线
   * @default true
   */
  hairline?: boolean;
  /**
   * 内容位置
   * @default 'center'
   */
  contentPosition?: DividerContentPosition;
}
