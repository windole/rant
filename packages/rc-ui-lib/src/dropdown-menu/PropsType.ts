import React, { ReactNode } from 'react';
import { BaseTypeProps, TeleportType } from '../utils';

type DirectionTypes = 'up' | 'down';

export type DropdownItemOption = {
  /** 文案 */
  text?: string;
  /** 标识符 */
  value?: string | number;
  /** 左侧图标名称或图片链接 */
  icon?: string | ReactNode;
};

export interface DropdownMenuProps extends BaseTypeProps {
  /** 菜单标题和选项的选中态颜色 */
  activeColor?: string;
  /** 菜单展开方向 */
  direction?: DirectionTypes;
  /** 菜单栏 z-index 层级 */
  zIndex?: string | number;
  /** 动画时长，单位秒 */
  duration?: number | string;
  /** 是否显示遮罩层 */
  overlay?: boolean;
  /** 是否在点击遮罩层后关闭菜单 */
  closeOnClickOverlay?: boolean;
  /** 是否在点击外部元素后关闭菜单 */
  closeOnClickOutside?: boolean;
  /** 子元素 */
  children?: React.ReactNode;
  /** 点击选项导致 value 变化时触发 */
  onChange?: (value: DropdownItemOption) => void;
  /** @private */
  shouldRenderTitle?: boolean;
}

export interface DropdownItemProps extends BaseTypeProps {
  /** 标志符 */
  name?: string;
  /** 菜单项标题 */
  title?: string;
  /** 默认值 */
  defaultValue?: DropdownItemOption;
  /** 当前选中项对应的 value */
  value?: DropdownItemOption;
  /** 占位文本 */
  placeholder?: string;
  /** 选项数组 */
  options?: DropdownItemOption[];
  /** 是否禁用菜单 */
  disabled?: boolean;
  /** 标题额外类名 */
  titleClass?: string;
  /** 子元素 */
  children?: React.ReactNode;
  /** @private */
  showPopup?: boolean;
  /** 指定挂载的节点 */
  teleport?: TeleportType;
  /** 点击选项导致 value 变化时触发 */
  onChange?: (value: DropdownItemOption) => void;
  /** 打开菜单栏时触发 */
  onOpen?: () => void;
  /** 打开菜单栏且动画结束后触发 */
  onOpened?: () => void;
  /** 关闭菜单栏时触发 */
  onClose?: () => void;
  /** 关闭菜单栏且动画结束后触发 */
  onClosed?: () => void;
}

export interface DropdownMenuContextProps {
  props?: React.PropsWithChildren<DropdownMenuProps>;
  value?: Record<string, number | string>;
  offset?: number | string;
  /** 点击选项导致 value 变化时触发 */
  onChange?: (obj) => void;
  renderTitle?: (ins: DropdownItemInstance) => void;
  close?: () => void;
}

export interface DropdownMenuInstance {
  close: () => void;
}
export interface DropdownItemInstance {
  /** 切换菜单展示状态，传 true 为显示，false 为隐藏，不传参为取反	 */
  toggle: (
    show?: boolean,
    options?: {
      immediate?: boolean;
    },
  ) => void;
  /** @private */
  state: {
    showPopup: boolean;
    transition: boolean;
    showWrapper: boolean;
  };
  /** @private */
  renderTitle: (value) => string | ReactNode;
}
