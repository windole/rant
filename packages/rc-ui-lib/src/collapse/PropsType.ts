import { ReactNode } from 'react';
import { CellProps } from '../cell/PropsType';
import { BaseTypeProps } from '../utils';

export interface CollapseProps extends BaseTypeProps {
  /** 是否开启手风琴模式	 */
  accordion?: boolean;
  /** 默认打开的面板name */
  initValue?: string | number | Array<number | string>;
  /** 是否显示外边框	 */
  border?: boolean;
  /** 当前展开面板的 name	 */
  value?: number | string | Array<number | string>;
  /** 子组件	 */
  children?: ReactNode;
  /** 切换面板时触发	*/
  onChange?: (activeNames: number | string | Array<number | string>) => void;
}

export interface CollapseItemProps extends CellProps {
  /** 唯一标识符，默认为索引值	 */
  name?: number | string;
  /** 是否禁用面板	 */
  disabled?: boolean;
  /** 是否为只读状态，只读状态下无法操作面板	 */
  readonly?: boolean;
  /** @private */
  index?: number;
  /** 子组件	 */
  children?: ReactNode;
  // onClick: () => void;
}

export type CollapseItemInstance = {
  toggle: (expand: boolean) => void;
};
