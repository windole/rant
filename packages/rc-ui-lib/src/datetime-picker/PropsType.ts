import { PickerInstance, PickerProps } from '../picker/PropsType';

export type DatetimePickerColumnType = 'year' | 'month' | 'day' | 'hour' | 'minute';

export type DatetimePickerType =
  | 'date'
  | 'time'
  | 'datetime'
  | 'datehour'
  | 'month-day'
  | 'year-month';

export interface SharedProps extends Omit<PickerProps, 'onConfirm' | 'onChange' | 'columns'> {
  value?: string | Date;
  /** 选项过滤函数	 */
  filter?: (type: string, values: string[]) => string[];
  /**
   * 自定义列排序数组,
   * 子项可选值为: year、month、day、hour、minute
   */
  columnsOrder?: string[];
  /** 选项格式化函数	*/
  formatter?: (type: string, value: string) => string;
  /** 点击完成按钮时触发的事件	 */
  onConfirm?: (value: Date) => void;
  /** 当值变化时触发的事件	 */
  onChange?: (value: Date) => void;
  onCancel?: () => void;
}

export interface DatePickerProps extends SharedProps {
  /** 时间类型 */
  type?: DatetimePickerType;
  /** 可选的最小时间，精确到分钟	 */
  minDate?: Date;
  /** 可选的最大时间，精确到分钟	 */
  maxDate?: Date;
}

export interface TimePickerProps extends SharedProps {
  /** 可选的最小小时	 */
  minHour?: number | string;
  /** 可选的最大小时	 */
  maxHour?: number | string;
  /** 可选的最小分钟	 */
  minMinute?: number | string;
  /** 可选的最大分钟	 */
  maxMinute?: number | string;
}

export type DateTimePickerProps = DatePickerProps & TimePickerProps;

export type DateTimePickerInstance = {
  getPicker: () => PickerInstance;
};
