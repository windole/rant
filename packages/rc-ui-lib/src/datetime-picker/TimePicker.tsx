import React, {
  forwardRef,
  useRef,
  useState,
  useMemo,
  useEffect,
  useImperativeHandle,
} from 'react';
import { Picker } from '..';
import { DateTimePickerInstance, TimePickerProps } from './PropsType';
import { PickerInstance } from '../picker';
import { useUpdateEffect } from '../hooks';
import { padZero, range } from '../utils';
import { times } from './utils';

const TimePicker = forwardRef<DateTimePickerInstance, TimePickerProps>((props, ref) => {
  const formatValue = (value?: string): string => {
    const { minHour, maxHour, minMinute, maxMinute } = props;

    if (!value) {
      value = `${padZero(minHour)}:${padZero(minHour)}`;
    }

    let [hour, minute] = value.split(':');
    hour = padZero(range(+hour, +minHour, +maxHour));
    minute = padZero(range(+minute, +minMinute, +maxMinute));

    return `${hour}:${minute}`;
  };

  const picker = useRef<PickerInstance>(null);
  const [currentDate, setCurrentDate] = useState(formatValue(props.value as string));
  const currentDateRef = useRef<string>(null);
  currentDateRef.current = currentDate;

  const ranges = useMemo(
    () => [
      {
        type: 'hour',
        range: [+props.minHour, +props.maxHour],
      },
      {
        type: 'minute',
        range: [+props.minMinute, +props.maxMinute],
      },
    ],
    [props.minHour, props.maxHour, props.minMinute, props.maxMinute],
  );

  const originColumns = useMemo(
    () =>
      ranges.map(({ type, range: rangeArr }) => {
        let values = times(rangeArr[1] - rangeArr[0] + 1, (index) => padZero(rangeArr[0] + index));

        if (props.filter) {
          values = props.filter(type, values);
        }

        return {
          type,
          values,
        };
      }),
    [ranges, props.filter],
  );

  const columns = useMemo(
    () =>
      originColumns.map((column) => ({
        values: column.values.map((value) => props.formatter(column.type, value)),
      })),
    [originColumns],
  );

  const updateColumnValue = (date?: string) => {
    const pair = date ? date.split(':') : currentDate.split(':');
    const values = [props.formatter('hour', pair[0]), props.formatter('minute', pair[1])];

    setTimeout(() => {
      picker.current?.setValues(values);
    }, 0);
  };

  const updateInnerValue = () => {
    const [hourIndex, minuteIndex] = picker.current.getIndexes();
    const [hourColumn, minuteColumn] = originColumns;

    const hour = hourColumn.values[hourIndex] || hourColumn.values[0];
    const minute = minuteColumn.values[minuteIndex] || minuteColumn.values[0];

    setCurrentDate(formatValue(`${hour}:${minute}`));
    updateColumnValue(formatValue(`${hour}:${minute}`));
  };

  const onConfirm = () => props.onConfirm?.(currentDate as unknown as Date);
  const onCancel = () => props.onCancel();

  const onChange = (_) => {
    updateInnerValue();
    setTimeout(() => {
      props.onChange?.(currentDateRef as unknown as Date);
    }, 0);
  };

  useEffect(() => {
    updateColumnValue();
  }, [columns]);

  useUpdateEffect(() => {
    updateInnerValue();
  }, [props.filter, props.maxHour, props.minMinute, props.maxMinute]);

  useUpdateEffect(() => {
    setTimeout(updateInnerValue, 0);
  }, [props.minHour]);

  useUpdateEffect(() => {
    const value = formatValue(props.value as string);

    if (value !== currentDate) {
      setCurrentDate(value);
      updateColumnValue(value);
    }
  }, [props.value]);

  useImperativeHandle(ref, () => ({
    getPicker: () => picker.current,
  }));

  return (
    <Picker
      ref={picker}
      columns={columns}
      readonly={props.readonly}
      onChange={onChange}
      onCancel={onCancel}
      onConfirm={onConfirm}
    />
  );
});

TimePicker.defaultProps = {
  minHour: 0,
  maxHour: 23,
  minMinute: 0,
  maxMinute: 59,
  formatter: (type: string, value: string) => value,
};

export default TimePicker;
