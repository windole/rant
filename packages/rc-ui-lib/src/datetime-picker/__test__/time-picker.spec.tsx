import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import TestsEvent from '../../../tests/events';
import { sleep } from '../../../tests/utils';
import TimePicker from '../TimePicker';
import { DateTimePickerInstance, TimePickerProps } from '../PropsType';

function filter(type: string, options: string[]): string[] {
  const mod = type === 'minute' ? 10 : 5;
  return options.filter((option) => Number(option) % mod === 0);
}

function formatter(type: string, value: string): string {
  return `${value} ${type}`;
}

describe('DateTimePicker', () => {
  function createTimePicker(props?: Partial<TimePickerProps>) {
    const pickerRef = React.createRef<DateTimePickerInstance>();

    const { queryByTestId, container, rerender, debug } = render(
      <TimePicker ref={pickerRef} {...props} />,
    );

    return {
      container,
      rerender,
      debug,
      queryByTestId,
      pickerRef,
    };
  }

  afterEach(() => {
    cleanup();
    jest.restoreAllMocks();
  });

  it('format initial value', () => {
    const { container } = createTimePicker({
      minHour: 22,
      minMinute: 58,
    });

    expect(container).toMatchSnapshot();
  });

  it('max-hour & max-minute', () => {
    const { container } = createTimePicker({
      value: '23:59',
      maxHour: 2,
      maxMinute: 2,
    });

    expect(container).toMatchSnapshot();
  });

  it('filter prop', () => {
    const { container } = createTimePicker({
      value: '12:00',
      filter,
    });

    expect(container).toMatchSnapshot();
  });

  it('formatter prop', async () => {
    const onConfirm = jest.fn();

    const { container, pickerRef } = createTimePicker({
      value: '12:00',
      filter,
      formatter,
      onConfirm,
    });
    expect(container).toMatchSnapshot();

    const confirmBtn = container.querySelector('.rc-picker__confirm');
    const columnWrapper = container.querySelector('.rc-picker-column');
    await TestsEvent.triggerDrag(columnWrapper, [0, -100]);
    await sleep(100);
    fireEvent.click(confirmBtn);

    expect(pickerRef.current.getPicker().getValues()).toEqual(['20 hour', '00 minute']);
  });

  it('should emit confirm event after clicking the confirm button', async () => {
    const onConfirm = jest.fn();

    const { container } = createTimePicker({
      onConfirm,
      value: '12:00',
    });

    const columnWrapper = container.querySelector('.rc-picker-column');
    await TestsEvent.triggerDrag(columnWrapper, [0, -300]);

    const confirmBtn = container.querySelector('.rc-picker__confirm');
    fireEvent.click(confirmBtn);

    expect(onConfirm.mock.calls[0][0]).toEqual('23:00');
  });

  it('should emit cancel event after clicking the cancel button', async () => {
    const onCancel = jest.fn();

    const { container } = createTimePicker({
      onCancel,
    });

    const cancelBtn = container.querySelector('.rc-picker__cancel');
    fireEvent.click(cancelBtn);

    expect(onCancel).toHaveBeenCalledTimes(1);
  });

  it('dynamic set value', async () => {
    const onConfirm = jest.fn();

    const { container, rerender } = createTimePicker({
      onConfirm,
    });

    await sleep(100);

    const props: TimePickerProps = {
      value: '00:00',
      onConfirm,
    };
    rerender(<TimePicker {...props} />);

    await sleep(100);

    const confirmBtn = container.querySelector('.rc-picker__confirm');
    fireEvent.click(confirmBtn);

    expect(onConfirm.mock.calls[0][0]).toEqual('00:00');
  });

  it('should emit value correctly when dynamic change minMinute', async () => {
    const onConfirm = jest.fn();

    const { container, rerender } = createTimePicker({
      onConfirm,
      value: '12:00',
      minMinute: 0,
    });

    await sleep(100);

    const props: TimePickerProps = {
      minMinute: 30,
      value: '12:00',
      onConfirm,
    };
    rerender(<TimePicker {...props} />);

    await sleep(100);

    const confirmBtn = container.querySelector('.rc-picker__confirm');
    fireEvent.click(confirmBtn);

    await sleep(100);

    expect(onConfirm.mock.calls[0][0]).toEqual('12:30');
  });

  it('should emit value correctly when dynamic change minHour', async () => {
    const onConfirm = jest.fn();

    const { container, rerender } = createTimePicker({
      onConfirm,
      value: '12:00',
      minHour: 0,
    });

    await sleep(100);

    const props: TimePickerProps = {
      minHour: 15,
      value: '12:00',
      onConfirm,
    };
    rerender(<TimePicker {...props} />);

    await sleep(100);

    const confirmBtn = container.querySelector('.rc-picker__confirm');
    fireEvent.click(confirmBtn);

    await sleep(100);

    expect(onConfirm.mock.calls[0][0]).toEqual('15:00');
  });

  it('set max-hour & max-minute smaller than current then emit correct value', async () => {
    const onConfirm = jest.fn();

    const { container, rerender } = createTimePicker({
      onConfirm,
      value: '23:59',
    });

    await sleep(100);

    const props: TimePickerProps = {
      maxHour: 2,
      maxMinute: 2,
      onConfirm,
    };
    rerender(<TimePicker {...props} />);

    await sleep(100);

    const confirmBtn = container.querySelector('.rc-picker__confirm');
    fireEvent.click(confirmBtn);

    expect(onConfirm.mock.calls[0][0]).toEqual('00:00');
  });
});
