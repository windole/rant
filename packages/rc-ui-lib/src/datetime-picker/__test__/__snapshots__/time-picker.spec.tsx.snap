// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimePicker filter prop 1`] = `
<div>
  <div
    class="rc-picker"
  >
    <div
      class="rc-picker__toolbar"
    >
      <button
        class="rc-picker__cancel"
        type="button"
      >
        取消
      </button>
      <button
        class="rc-picker__confirm"
        type="button"
      >
        确认
      </button>
    </div>
    <div
      class="rc-picker__columns"
      style="height: 220px;"
    >
      <div
        class="rc-picker-column"
      >
        <ul
          aria-hidden="true"
          class="rc-picker-column__wrapper"
          style="transition-duration: 0ms; transition-property: none; transform: translate3d(0,88px,0);"
        >
          <li
            class="rc-picker-column__item rc-picker-column__item--selected"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              00
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              05
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              10
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              15
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              20
            </div>
          </li>
        </ul>
      </div>
      <div
        class="rc-picker-column"
      >
        <ul
          aria-hidden="true"
          class="rc-picker-column__wrapper"
          style="transition-duration: 0ms; transition-property: none; transform: translate3d(0,88px,0);"
        >
          <li
            class="rc-picker-column__item rc-picker-column__item--selected"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              00
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              10
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              20
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              30
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              40
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              50
            </div>
          </li>
        </ul>
      </div>
      <div
        class="rc-picker__mask"
        style="background-size: 100% 88px;"
      />
      <div
        class="rc-hairline-unset--top-bottom rc-picker__frame"
        style="height: 44px;"
      />
    </div>
  </div>
</div>
`;

exports[`DateTimePicker format initial value 1`] = `
<div>
  <div
    class="rc-picker"
  >
    <div
      class="rc-picker__toolbar"
    >
      <button
        class="rc-picker__cancel"
        type="button"
      >
        取消
      </button>
      <button
        class="rc-picker__confirm"
        type="button"
      >
        确认
      </button>
    </div>
    <div
      class="rc-picker__columns"
      style="height: 220px;"
    >
      <div
        class="rc-picker-column"
      >
        <ul
          aria-hidden="true"
          class="rc-picker-column__wrapper"
          style="transition-duration: 0ms; transition-property: none; transform: translate3d(0,88px,0);"
        >
          <li
            class="rc-picker-column__item rc-picker-column__item--selected"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              22
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              23
            </div>
          </li>
        </ul>
      </div>
      <div
        class="rc-picker-column"
      >
        <ul
          aria-hidden="true"
          class="rc-picker-column__wrapper"
          style="transition-duration: 0ms; transition-property: none; transform: translate3d(0,88px,0);"
        >
          <li
            class="rc-picker-column__item rc-picker-column__item--selected"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              58
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              59
            </div>
          </li>
        </ul>
      </div>
      <div
        class="rc-picker__mask"
        style="background-size: 100% 88px;"
      />
      <div
        class="rc-hairline-unset--top-bottom rc-picker__frame"
        style="height: 44px;"
      />
    </div>
  </div>
</div>
`;

exports[`DateTimePicker formatter prop 1`] = `
<div>
  <div
    class="rc-picker"
  >
    <div
      class="rc-picker__toolbar"
    >
      <button
        class="rc-picker__cancel"
        type="button"
      >
        取消
      </button>
      <button
        class="rc-picker__confirm"
        type="button"
      >
        确认
      </button>
    </div>
    <div
      class="rc-picker__columns"
      style="height: 220px;"
    >
      <div
        class="rc-picker-column"
      >
        <ul
          aria-hidden="true"
          class="rc-picker-column__wrapper"
          style="transition-duration: 0ms; transition-property: none; transform: translate3d(0,88px,0);"
        >
          <li
            class="rc-picker-column__item rc-picker-column__item--selected"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              00 hour
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              05 hour
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              10 hour
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              15 hour
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              20 hour
            </div>
          </li>
        </ul>
      </div>
      <div
        class="rc-picker-column"
      >
        <ul
          aria-hidden="true"
          class="rc-picker-column__wrapper"
          style="transition-duration: 0ms; transition-property: none; transform: translate3d(0,88px,0);"
        >
          <li
            class="rc-picker-column__item rc-picker-column__item--selected"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              00 minute
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              10 minute
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              20 minute
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              30 minute
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              40 minute
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              50 minute
            </div>
          </li>
        </ul>
      </div>
      <div
        class="rc-picker__mask"
        style="background-size: 100% 88px;"
      />
      <div
        class="rc-hairline-unset--top-bottom rc-picker__frame"
        style="height: 44px;"
      />
    </div>
  </div>
</div>
`;

exports[`DateTimePicker max-hour & max-minute 1`] = `
<div>
  <div
    class="rc-picker"
  >
    <div
      class="rc-picker__toolbar"
    >
      <button
        class="rc-picker__cancel"
        type="button"
      >
        取消
      </button>
      <button
        class="rc-picker__confirm"
        type="button"
      >
        确认
      </button>
    </div>
    <div
      class="rc-picker__columns"
      style="height: 220px;"
    >
      <div
        class="rc-picker-column"
      >
        <ul
          aria-hidden="true"
          class="rc-picker-column__wrapper"
          style="transition-duration: 0ms; transition-property: none; transform: translate3d(0,88px,0);"
        >
          <li
            class="rc-picker-column__item rc-picker-column__item--selected"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              00
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              01
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              02
            </div>
          </li>
        </ul>
      </div>
      <div
        class="rc-picker-column"
      >
        <ul
          aria-hidden="true"
          class="rc-picker-column__wrapper"
          style="transition-duration: 0ms; transition-property: none; transform: translate3d(0,88px,0);"
        >
          <li
            class="rc-picker-column__item rc-picker-column__item--selected"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              00
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              01
            </div>
          </li>
          <li
            class="rc-picker-column__item"
            role="button"
            style="height: 44px;"
            tabindex="0"
          >
            <div
              class="rc-ellipsis"
            >
              02
            </div>
          </li>
        </ul>
      </div>
      <div
        class="rc-picker__mask"
        style="background-size: 100% 88px;"
      />
      <div
        class="rc-hairline-unset--top-bottom rc-picker__frame"
        style="height: 44px;"
      />
    </div>
  </div>
</div>
`;
